### PR TITLE
64 page 3 restructure page so drill down table is less lost

### PR DESCRIPTION
**Pull request by axcooper**
## Pull request overview

The table with subjects in was below the bar chart and easily missed, now they are both tabbed, so the table is more visible.
Data download tab added.
Messages added where no data.
Tier 2 subject area and standard names added to table.

## Pull request checklist

Please check if your PR fulfills the following:
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Tests have been run locally and are passing (`shinytest2::test_app()`)
- [x] Code is styled according to tidyverse styling (checked locally with `styler::style_dir()` and `lintr::lint_dir()`)

## What is the current behaviour?

The table with subjects in was below the bar chart and easily missed, if the user did not scroll down the page fully.

## What is the new behaviour?

The subjects bar chart is now a tab, along with the table, so the table is more visible.

The table has tier 2 subject area and standard name added, which were essential info.

Messages have been added to the provider tab where there are none of the measure selected for that year, and the bar chart is then blank. Currently the table will still have zeros in where there is eg no achievements for a provider in a given year but there are starts/enrolments. This may be useful as it shows the areas covered by the provider, but need feedback on this.

Dropdown of apprenticeship level(s) added, which helps to hone in on a standard. Message added to remind to select a level.

## Anything else

May also want to put in dropdown of standard name?
At the moment it is hard to hone in on a standard and find the providers for it.
It is OK selecting providers and seeing what standards they provide.

### CODE REVIEW

**Comment by axcooper** [comment]
Once this is merged in, will work on Page 3: Add individual apprenticeship standards breakdowns #72

File: R/dashboard_modules/03-subjects_and_standards.R [hunk 1]
```diff
@@ -199,5 +264,30 @@ subject_standards_server <- function(id) {
...
+        paste0(tolower(gsub(" ", "", raw_name)), extension)
+      },
+      ## Generate downloaded file ---------------------------------------------
+      content = function(file) {
+        if (input$file_type == "CSV (Up to 13.18 MB)") {
+          data.table::fwrite(subject_area_data(), file)
+        } else {
+          # Added a basic pop up notification as the Excel file can take time to generate
+          pop_up <- showNotification("Generating download file", duration = NULL)
+          openxlsx::write.xlsx(chars_reactive_table(), file, colWidths = "Auto")
```
Comment (cjrace) [inline R/dashboard_modules/03-subjects_and_standards.R:287]: I think `chars_reactive_table()` should be `subject_area_data()` here?

**Comment by axcooper** [comment]
Have taken out the bar chart and put in dropdown for standard.

Still to do: 

This is a must: 
Ensure that standards available in the dropdown correspond to the level(s) selected

This is a nice to have:
The provider and the standards table cross select with each other ( I have left the code that did this for the bar chart with the hope it could be used)

**Comment by cjrace** [comment]
Have jumped on this, updated it so the standards selection is dependent on the level selection - will look at the cross-filtering later today

**Comment by axcooper** [comment]
The selectable providers list is great, just need to be able to do the same going the other way...

**Comment by cjrace** [comment]
Not sure what the error is in the tests, I can't reproduce locally, so going to force the merge and check for errors on the deployed app

**Pull request merged**